java/client/src/main/java/glide/api/commands/servermodules/FT.java

```diff
@@ -703,6 +703,116 @@ public static CompletableFuture<String> aliasupdate(
+     * //    0
+     * //    10
+     * // }
+     * }</pre>
+     */
+    public static CompletableFuture<GlideString> explain(
+            @NonNull BaseClient client, @NonNull GlideString indexName, @NonNull GlideString query) {
+        var args = concatenateArrays(new GlideString[] {gs("FT.EXPLAIN"), indexName, query});
+        return executeCommand(client, args, false);
+    }
+
+    /**
+     * Same as the {@link FT#explain(BaseClient, String, String)} except that the results are
+     * displayed in a different format. More useful with cli.
+     *
+     * @param client The client to execute the command.
+     * @param indexName The index name to search into.
+     * @param query The text query to search. It is the same as the query passed as an argument to
+     *     {@link FT#search(BaseClient, String, String)}.
+     * @return A <code>String[[</code> representing the execution plan.
+     * @example
+     *     <pre>{@code
+     * FT.explaincli(client, "myIndex",  "@price:[0 10]").get();
+     * // the output can look like this (the result is an array)
+     * //  Field {
+     * //    price
+     * //    0
+     * //    10
+     * // }
+     * }</pre>
+     */
+    public static CompletableFuture<String[]> explaincli(
+            @NonNull BaseClient client, @NonNull String indexName, @NonNull String query) {
+        CompletableFuture<GlideString[]> result = explaincli(client, gs(indexName), gs(query));
+        return result.thenApply(
+                ret -> Arrays.stream(ret).map(e -> e.getString()).toArray(String[]::new));
+    }
+
+    /**
+     * Same as the {@link FT#explain(BaseClient, String, String)} except that the results are
+     * displayed in a different format. More useful with cli.
+     *
+     * @param client The client to execute the command.
+     * @param indexName The index name to search into.
+     * @param query The text query to search. It is the same as the query passed as an argument to
+     *     {@link FT#search(BaseClient, String, String)}.
+     * @return A <code>GlideString[]</code> representing the execution plan.
+     * @example
+     *     <pre>{@code
+     * FT.explaincli(client, "myIndex",  "@price:[0 10]").get();
+     * // the output can look like this (the result is an array)
+     * //  Field {
+     * //    price
+     * //    0
+     * //    10
+     * // }
+     * }</pre>
+     */
+    public static CompletableFuture<GlideString[]> explaincli(
+            @NonNull BaseClient client, @NonNull GlideString indexName, @NonNull GlideString query) {
+        var args = concatenateArrays(new GlideString[] {gs("FT.EXPLAINCLI"), indexName, query});
+        CompletableFuture<GlideString[]> result =
+                ((GlideClusterClient) client)
+                        .customCommand(args)
+                        .thenApply(ClusterValue::getSingleValue)
+                        .thenApply(ret -> (Object[]) ret)
+                        .thenApply(ret -> castArray(ret, GlideString.class));
```

Comment:
Reuse existing `executeCommand` and add apply `castArray` to the reposnse

Reply:
I think we've tried that when we were developing and there was an issue with the return type ..

@acarbonetto

Reply:
So need to resolve that. Please don't duplicate the code, don't cast client to cluster one - it could be standalone.